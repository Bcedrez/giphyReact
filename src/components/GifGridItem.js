import React from "react";
import PropTypes from 'prop-types';


const GifGridItem = ({id,titulo,url})=>{
    // console.log({id,titulo,url})
    return (
        <div className="card animate__animated animate__zoomIn">
           <img src={url} alt={titulo} />
           <p>{titulo}</p>
        </div>
    );
}

GifGridItem.propTypes = {
    titulo: PropTypes.string.isRequired,
    url : PropTypes.string.isRequired
}


export default GifGridItem