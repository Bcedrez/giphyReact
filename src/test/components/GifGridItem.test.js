import React from 'react';
import { shallow } from 'enzyme';




import GifGridItem from '../../components/GifGridItem'

describe('Prueba sobre <GifGridItem />',()=>{

    const titulo = 'Un Titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem url={url} titulo={titulo} /> );

    test('Se debe renderizar correctamente',()=>{       
        
        expect(wrapper).toMatchSnapshot();        

    });

    test('Debe tener un parrafo con el titulo',()=>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo);
    });

    test('Debe de tener imagen src = url y alt = tiulo',()=>{
        const imagen = wrapper.find('img');
        const {src,alt} = imagen.props();

        expect(src).toBe(url);
        expect(alt).toBe(titulo);
    });

    test('div debe tener animacion animate__zoomIn',()=>{
        const div = wrapper.find('div');        
        const {className} = div.props();        
        expect(className.includes('animate__zoomIn')).toBe(true);
    })

})

