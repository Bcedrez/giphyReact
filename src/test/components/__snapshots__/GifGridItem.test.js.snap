// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre <GifGridItem /> Se debe renderizar correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Un Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
